syntax = "proto2";
package actions.proto;

import "util/proto/object_id.proto";

// Simple test strategies, not meant for serious use.
// Buy apples in Appleville, carry to Orangetown, sell apples, buy oranges,
// go back and sell the oranges, repeat.
message ShuttleTrade {
  optional string good_a = 1;
  optional string good_z = 2;
  optional util.proto.ObjectId area_a_id = 3;
  optional util.proto.ObjectId area_z_id = 4;
  enum State {
    STS_BUY_A = 0;
    STS_BUY_Z = 1;
  }
  optional State state = 5;
  optional uint64 tolerance = 6 [default = 900000];
}

// Strategies for SevenYears naval-management minigame.
message SevenYearsMerchant {
  optional util.proto.ObjectId base_area_id = 1;
  optional string mission = 2;
  optional string default_mission = 3;
}

message Strategy {
  oneof strategy {
    ShuttleTrade shuttle_trade = 1;
    SevenYearsMerchant seven_years_merchant = 2;
  }
  oneof key {
    // If 'define' is set, units may copy this strategy by setting
    // 'lookup' to that string.
    string define = 3;
    string lookup = 4;
  }
}

